Remove return values for asserts

samuelsinayoko opened the pull request:
Unless they are context managers.

- [X] closes #25135 
- [X] tests added / passed (not applicable)
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

samuelsinayoko committed:
Remove return values for asserts

Unless they are context managers.

WillAyd commented:
Looks like a slew of test changes as a result of code like `assert tm.assert_X_equal` which should get updated here.

If you test locally first should make it easier to find up front

samuelsinayoko commented:
OK am hoping to fix the tests shortly.

samuelsinayoko committed:
Merge branch 'master' into 25135

samuelsinayoko committed:
Dont assert functions that assert

assert either raises or doesn't. If it doesn't tm.assert_* returns None,
and there's no need to assert XXX is None.

samuelsinayoko committed:
flake8

samuelsinayoko committed:
testing.util.is_sorted raises AssertionError

samuelsinayoko committed:
Improve is_sorted docstring

gfyoung commented:
@samuelsinayoko : This looks much better!  I still see a handful of `return assert_*` statements in that file (`assert_almost_equal` and `assert_series_equal` implementations contain the remainder).  Can those be removed as well?

samuelsinayoko committed:
remove remaining return assert_* statements

samuelsinayoko commented:
> @samuelsinayoko : This looks much better! I still see a handful of `return assert_*` statements in that file (`assert_almost_equal` and `assert_series_equal` implementations contain the remainder). Can those be removed as well?

done in 60ec5ee

gfyoung commented:
@samuelsinayoko : I think you only removed the ones in `assert_almost_equal`.  There are still two in `assert_series_equal`.  Were you able to remove those?

samuelsinayoko commented:
Ha sorry! Will update the PR today.

Sent from my iPhone

> On 4 Mar 2019, at 08:48, gfyoung <notifications@github.com> wrote:
> 
> @samuelsinayoko : I think you only removed the ones in assert_almost_equal. There are still two in assert_series_equal. Were you able to remove those?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

samuelsinayoko committed:
remove remaining "return assert_*" statements in testing.py

samuelsinayoko committed:
Fix broken tests

gfyoung commented on pandas/util/testing.py:
```diff
@@ -1324,11 +1323,13 @@ def assert_series_equal(left, right, check_dtype=True,
         assert is_extension_array_dtype(right.dtype)
-        return assert_extension_array_equal(left._values, right._values)
+        assert_extension_array_equal(left._values, right._values)
+        return
```
Hmm...not sure if this is ideal, but at least it makes it clear `assert_extension_array_equal` doesn't return anything.  Could potentially revisit in a follow-up.

WillAyd replied:
Is there actually a reason to return here? Seems like it would unintentionally skip out on subsequent checks

gfyoung replied:
@WillAyd : Not sure, but let's revisit in a follow-up, since the goal is to just refactor.

WillAyd replied:
Yea sure makes sense @samuelsinayoko if you'd like to tackle

jreback replied:
you can leave out the return statement. assert returns None which is the same

WillAyd reviewed (approved):
lgtm @jreback

jreback commented on pandas/tests/series/test_constructors.py:
```diff
@@ -1009,7 +1009,7 @@ def test_fromDict(self):
 
         series = Series(data)
-        assert tm.is_sorted(series.index)
+        tm.is_sorted(series.index)
```
can you rename this routine to tm.assert_is_sorted

jreback commented on pandas/util/testing.py:
```diff
@@ -1324,11 +1323,13 @@ def assert_series_equal(left, right, check_dtype=True,
           is_extension_array_dtype(right) and not is_categorical_dtype(right)):
-        return assert_extension_array_equal(left.array, right.array)
+        assert_extension_array_equal(left.array, right.array)
+        return
```
same don't need a return

samuelsinayoko replied:
I tried this earlier but some of the tests failed so I added the returns back.

samuelsinayoko replied:
See https://github.com/pandas-dev/pandas/pull/25462/commits/091a00e2aa5a7bb81a16731ad77ca11405e514d6 (this was in response to an earlier PR comment)

jreback commented:
can you merge master and update

samuelsinayoko committed:
Rename is_sorted to assert_is_sorted

samuelsinayoko committed:
Merge branch 'master' of https://github.com/pandas-dev/pandas into 25135

samuelsinayoko commented:
@jreback have merged master and updated the branch. Don't think I can leave out the return statements as you suggested because that breaks the buil (search back for the build of 091a00e2 https://github.com/pandas-dev/pandas/commit/091a00e2aa5a7bb81a16731ad77ca11405e514d6 above)

jreback commented:
@samuelsinayoko you may need to fix some tests that are actually asserting the result of this assert, but I want to remove the return values

samuelsinayoko commented:
OK I'll give this another go.

On Tue, 26 Mar 2019 at 11:52, Jeff Reback <notifications@github.com> wrote:

> @samuelsinayoko <https://github.com/samuelsinayoko> you may need to fix
> some tests that are actually asserting the result of this assert, but I
> want to remove the return values
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/pandas-dev/pandas/pull/25462#issuecomment-476592213>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AQuwvoIm4H69yV_3a7pOHtrxIzmje36Rks5vagntgaJpZM4bVN-a>
> .
>


-- 

*Sam Sinayoko* | Data Scientist

*BMLL Technologies Ltd.*

10th Floor – Portland House

Bressenden Place

London

SW1E 5RS

+44(0)20 3828 9020

jreback commented:
can you merge master and update

samuelsinayoko commented:
Apologies, haven't found the time to come back to this as work's been crazy
for  the past couple of weeks but will update the PR over the weekend.

On Fri, 5 Apr 2019 at 01:51, Jeff Reback <notifications@github.com> wrote:

> can you merge master and update
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/pandas-dev/pandas/pull/25462#issuecomment-480112813>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AQuwvuUm4RYJbZPrU-0idZAykUR1dzw5ks5vdp4SgaJpZM4bVN-a>
> .
>


-- 

*Sam Sinayoko* | Data Scientist

*BMLL Technologies Ltd.*

10th Floor – Portland House

Bressenden Place

London

SW1E 5RS

+44(0)20 3828 9020

samuelsinayoko committed:
Merge remote-tracking branch 'upstream/master' into 25135

samuelsinayoko committed:
Merge branch '25135' of github.com:samuelsinayoko/pandas into 25135

jreback commented on pandas/util/testing.py:
```diff
@@ -1106,11 +1105,13 @@ def assert_series_equal(left, right, check_dtype=True,
           is_extension_array_dtype(right) and not is_categorical_dtype(right)):
-        return assert_extension_array_equal(left.array, right.array)
+        assert_extension_array_equal(left.array, right.array)
+        return
```
this just returns None which is the same as if you leave the return out, pls remove

samuelsinayoko committed:
reactivate metadata comparison

jreback commented:
can you merge master

WillAyd commented:
Can you merge and fix up conflicts with master? I think this will be a precursor to #26234

samuelsinayoko commented:
Sure. Will do tonight.

Sent from my iPhone

> On 30 Apr 2019, at 02:45, William Ayd <notifications@github.com> wrote:
> 
> Can you merge and fix up conflicts with master? I think this will be a precursor to #26234
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

samuelsinayoko committed:
Merge remote-tracking branch 'upstream/master' into 25135

jreback committed:
handle expected in integer array testing

jreback commented:
not exactly sure why, but this surfaced an error in our tests

pushed a patch, let's see

WillAyd commented:
Looks like a test failure in pandas/tests/io/parser/test_textreader.py which is inadvertently expecting a return value from one of the assert functions; if you can take a look

samuelsinayoko commented:
Will do.

On Wed, 1 May 2019 at 15:06, William Ayd <notifications@github.com> wrote:

> Looks like a test failure in pandas/tests/io/parser/test_textreader.py
> which is inadvertently expecting a return value from one of the assert
> functions; if you can take a look
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/pandas-dev/pandas/pull/25462#issuecomment-488291236>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEF3BPWUZSFDCWFM75YBYH3PTGPWNANCNFSM4G2U36NA>
> .
>


-- 

*Sam Sinayoko* | Data Scientist

*BMLL Technologies Ltd.*

10th Floor – Portland House

Bressenden Place

London

SW1E 5RS

+44(0)20 3828 9020

jreback committed:
Merge branch 'master' into PR_TOOL_MERGE_PR_25462

jreback committed:
fixed assert

jreback commented:
i pushed again, let's see

jreback committed:
Merge branch 'master' into PR_TOOL_MERGE_PR_25462

jreback committed:
fix test

jreback commented:
@WillAyd I think we would get a lot of mileage out of typing these assert_* functions, can you create an issue

jreback commented:
thanks for the patch @samuelsinayoko

WillAyd commented:
@jreback sounds good, though I was actually thinking we'd almost always ignore `pandas/tests` from static analysis. 

In any case opened #26302 so can discuss there as needed
